### PR TITLE
Prod 125 evm compatibility #2

### PR DESCRIPTION
**Pull request by m4sterbunny**
This recreates the edits on closed PR rather than fix merge conflicts

### CODE REVIEW

File: docs/evm_compatibility/json_rpc_api_methods.md [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
 approvedBy: na
-comment: 
+comments: TODOs inline 
 ---
 
-*The following tables present the full list of JSON-RPC methods. Each of them is in one of the states:*
+This page presents the full list of JSON-RPC methods. Each of them is in one of three states:
 
-|![done](img/done.ico)| — done|![in progress](img/inprogress.ico) | — in progress|![todo](img/todo.ico) | — to do|
+<!-- todo Andrey can help by providing the deviations fr0m ETH standard methods / Neon equivalent will send 1n Slack todo Neon methods missing -- needs update == Andrey?? -->
```
Comment (m4sterbunny) [inline docs/evm_compatibility/json_rpc_api_methods.md:12]: @afalaleev I am returning to the EVM compatibility section PTAL in the discussion you took on providing deviations from Eth standard methods. Is that still with you, or is there someone you want me to ping?

File: docs/evm_compatibility/json_rpc_api_methods.md [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
+
+|![done](img/done.ico)| — done|![in progress](img/inprogress.ico) | — in progress|![not supported](img/false-copy.png) | — not supported|
```
Comment (m4sterbunny) [inline docs/evm_compatibility/json_rpc_api_methods.md:14]: @afalaleev I seem to recall, but failed to make a note >> that all the "in progress" items were supposed to become "not supported". Please can you confirm my memory of this!

File: docs/evm_compatibility/overview.md [hunk 3]
```diff
@@ -1,37 +1,37 @@
 ---
 title: EVM Compatibility Overview
 proofedDate: na
-iterationBy: na
+iterationBy: HB
 includedInSite: true
 approvedBy: na
-comment: 404 being dealt with on link-fixer branch??
+comments: #1 we need a page on the not supported OpCodes Anton will send by slack #2 Requires supported precompiles Anton can provide by slack + explanation of WHY they are not supported #3 Requires a list of native precomiled contracts Anton can provide by slack [actually, the yellow paper https://ethereum.github.io/yellowpaper/paper.pdf is not a great source for the opcodes -- using docs instead]
```
Comment (m4sterbunny) [inline docs/evm_compatibility/overview.md:7]: @afalaleev 1. is dealt with above. 2. We were going to include something on the WHY not supported 3. We have precompiles to list still

File: docs/evm_compatibility/overview.md [hunk 4]
```diff
@@ -40,24 +40,39 @@ Interoperability between Solana and Ethereum EVMs does require certain adaptatio
...
 
 Several Solana-specific differences also impact smart contract development.
 
-### Upper limit on number of Accounts
-Neon EVM uses Solana Transaction V0 with an extension via [account lookup tables](https://docs.solana.com/developing/lookup-tables): this limits the maximum number of accounts to 64 in a single transaction. Solana requires that all accounts used in a transaction be specified in order to ensure parallel execution of transactions. 
+### Upper limit on number of accounts
+<!-- link to solana tx renamed and relinked -->
+Neon EVM uses [Solana Transaction V0](https://docs.solana.com/developing/versioned-transactions): limiting the maximum number of accounts used in a single transaction to 64. Solana requires that all accounts used in a transaction be specified in order; to ensure parallel execution of transactions.
+
+<!-- go deeper on HOW to modify the contract to constrain account numbers Anton will pass in slack
```
Comment (m4sterbunny) [inline docs/evm_compatibility/overview.md:60]: @afalaleev We have a note to provide boilerplate code to constrain account numbers

File: docs/evm_compatibility/overview.md [hunk 5]
```diff
@@ -40,24 +40,39 @@ Interoperability between Solana and Ethereum EVMs does require certain adaptatio
...
-To avoid the occurrence of a heap overflow error, it is recommended that you:
-* Reduce the size of the call stack
-* Reduce the number of variables used
+To avoid the occurrence of a heap overflow error, it's recommended that you reduce the size of the:
+- call stack
+- contract
 
-### Limitation on `block.timestamp` / `block.number` Usage
+<!-- support users further on HOW to reduce heap size?
+
```
Comment (m4sterbunny) [inline docs/evm_compatibility/overview.md:71]: @gigimon We have a note that you can provide information on troubleshooting to id when heap size is the cause of issues

Comment (gigimon) [reply docs/evm_compatibility/overview.md:71]: I can't describe this section :(

File: docs/evm_compatibility/overview.md (hunk 4 above)
Comment (m4sterbunny) [reply docs/evm_compatibility/overview.md:60]: @anton-lisanin -- sorry this is for you, not Andrey!

File: docs/evm_compatibility/overview.md (hunk 5 above)
Comment (m4sterbunny) [reply docs/evm_compatibility/overview.md:71]: @anton-lisanin was it you who could provide info on the heap size troubleshooting?

File: docs/evm_compatibility/overview.md (hunk 4 above)
Comment (m4sterbunny) [reply docs/evm_compatibility/overview.md:104]: some boilerplate code in place -- closing and relocating comments to there

File: docs/evm_compatibility/overview.md [hunk 6]
```diff
@@ -1,63 +1,155 @@
...
 
-Solidity or Vyper smart contracts, standard development and deployment tools and practices can all be applied on Neon EVM with minimal reconfiguration of the code:
+Solidity or Vyper smart contracts, standard development and deployment tools and practices can all be applied on Neon EVM with minimal reconfiguration of the code.
 
 - Neon EVM accounts follow [Ethereum's account standards](https://ethereum.org/en/developers/docs/accounts/) 
 
-> All [Ethereum opcodes](https://www.evm.codes/?fork=merge) are represented verbatim on Neon EVM. 
+>  The majority of [Ethereum OpCodes](https://ethereum.org/en/developers/docs/evm/opcodes) are represented verbatim on Neon EVM. Learn about stacks: 41, 44, 45, and 48 [OpCodes](opcodes).
+
+- Neon EVM supports type 0 / legacy transaction requests
```
Comment (m4sterbunny) [inline docs/evm_compatibility/overview.md:34]: @anton-lisanin I recall that there was not yet support for EIP-1559, is this still the case?

File: docs/evm_compatibility/overview.md [hunk 7]
```diff
@@ -1,63 +1,155 @@
...
+
+ - Non-executable
+
+> Since executable accounts are immutable, their application state is stored in non-executable accounts.
+
+The differences go deeper still. In an Ethereum EVM, contracts can only read and write their own storage. In Sealevel, any account’s data can be read or written to by a contract. However, the runtime enforces that only an account’s “owner” is allowed to modify it. Changes by any other programs will be reverted and cause the transaction to fail.
+
+:::info
+
+Learn more in the [Solana wiki](https://solana.wiki/docs/).
```
Comment (m4sterbunny) [inline docs/evm_compatibility/overview.md:88]: @anton-lisanin not sure who is responsible for upkeep of this wiki -- is this an authoritative resource, may I use it?

File: docs/evm_compatibility/overview.md [hunk 8]
```diff
@@ -1,63 +1,155 @@
...
+
+:::
+
+### Account mapping
+
+Each Ethereum account involved in a transaction must be mapped to a corresponding Solana account. Any call made to the Ethereum account (e.g. to read balance, execute a transaction, etc.) requires that the Solana account is included to make use of the storage provided.
 
-### Upper limit on number of Accounts
-Neon EVM uses Solana Transaction V0 with an extension via [account lookup tables](https://docs.solana.com/developing/lookup-tables): this limits the maximum number of accounts to 64 in a single transaction. Solana requires that all accounts used in a transaction be specified in order to ensure parallel execution of transactions. 
+<!-- based on item in Slack https://neonlabsworkspace.slack.com/archives/C03CQ8A6WTT/p1683107335962669 >> not sure that this is functional end user support as it stands -->
```
Comment (m4sterbunny) [inline docs/evm_compatibility/overview.md:96]: @anton-lisanin I did my best to make use of what was in Slack -- not convinced that this fully covers what the end-user may need.

File: docs/evm_compatibility/overview.md [hunk 9]
```diff
@@ -1,63 +1,155 @@
...
+/* 32 accounts: */
+mapping(uint256 => uint256) map;
+...
+
+for (uint256 i = 0; i < 32; i++) {
+   map[i] = i;
+}
+```
+
+<!-- todo looks like these code snippets will benefit from some better commentary -->
```
Comment (m4sterbunny) [inline docs/evm_compatibility/overview.md:129]: @anton-lisanin can you provide deeper information for comments in code -- or a deeer explanation of the scenario to which these 2 variants may be applied to?
Just a bit more context may assist the reader.

File: docs/evm_compatibility/overview.md [hunk 10]
```diff
@@ -1,63 +1,155 @@
...
+- Event emission
+	- Use indexed parameters
+	- Avoid strings, arrays, and mappings
+- Returning values from a function
+	- Don't return data you don't use!
+	- Avoid strings, arrays, and mappings
+- Local variables: avoid strings, arrays, and mappings
 
-### Limitation on `block.timestamp` / `block.number` Usage
+<!-- todo How do users know when the issue is a heap overflow error? Can we show logs ?screenshot and name of service? to demonstrate when the issue is heap size? == detection method ?? Oleg should be able to provide -->
```
Comment (m4sterbunny) [inline docs/evm_compatibility/overview.md:149]: @anton-lisanin Oleg could not provide anything -- when we had the meet, we talked about helping end users figure out when the issue could be attributed to heap overflow

File: docs/evm_compatibility/precompiles.md [hunk 11]
```diff
@@ -4,19 +4,21 @@ proofedDate: na
...
+Contracts written in the Solidity language will in Neon EVM if they contain calls to the following:
+* bigModExp — Used for efficient RSA verification inside an EVM, as well as other forms of number theory-based cryptography
+* bn256Add — Performs addition on the elliptic curve operations
+* bn256ScalarMult — Performs scalar multiplication on the elliptic curve operations
+* bn256Pairing — Elliptic curve pairing operations to perform zkSTARKs verification within the block gas limit
 
-The Neon EVM requires the implementation of system calls in Solana for these contracts.
+> The Neon EVM requires the implementation of system calls in Solana to support these contracts in the future.
+
+<!-- We have our own native Precompiled contracts -- need to list those and provide addresses -->
```
Comment (m4sterbunny) [inline docs/evm_compatibility/precompiles.md:24]: @anton-lisanin do we have a single source of truth for the precompiles that nEVM provides?

File: docs/evm_compatibility/overview.md (hunk 6 above)
Comment (anton-lisanin) [reply docs/evm_compatibility/overview.md:34]: It is still the case. We support only legacy transactions.

File: docs/evm_compatibility/overview.md (hunk 7 above)
Comment (anton-lisanin) [reply docs/evm_compatibility/overview.md:88]: Never seen this wiki before. I think we should not use it.

File: docs/evm_compatibility/overview.md (hunk 8 above)
Comment (anton-lisanin) [reply docs/evm_compatibility/overview.md:96]: Remove entire section? Merge it with "Upper limit on number of accounts"

File: docs/evm_compatibility/overview.md (hunk 9 above)
Comment (anton-lisanin) [reply docs/evm_compatibility/overview.md:129]: Code is extremely simple. There really nothing to comment.
Context was lost when moving from Slack

File: docs/evm_compatibility/overview.md (hunk 10 above)
Comment (anton-lisanin) [reply docs/evm_compatibility/overview.md:149]: Example transaction with heap overflow - https://neonscan.org/tx/0x6bdc50921eaa8981e2f534e09139840fd176bf092bd835393ee4bab998409c0a

Neonscan has a list of Solana transactions that was executed. And if open last Solana transaction in solscan, you could see error message in the logs. https://solscan.io/tx/3fuTXn8SiAwTrpG96VmhvuHQe9tssFv7y2yHFynVpM4d51uBZerehxZebPTFvKLs8b2wmnXmhxu5yhAwWdnUJLVC?cluster=devnet

File: docs/evm_compatibility/precompiles.md (hunk 11 above)
Comment (anton-lisanin) [reply docs/evm_compatibility/precompiles.md:24]: Not yet. And should be separate page. Not related to compatibility

File: docs/evm_compatibility/overview.md (hunk 9 above)
Comment (m4sterbunny) [reply docs/evm_compatibility/overview.md:129]: @anton-lisanin We need just 1 sentence describing the difference between the first and second examples -- why one is preferable to the other @YYYurchenko agrees we need a bit of context here.

File: docs/evm_compatibility/overview.md (hunk 10 above)
Comment (m4sterbunny) [reply docs/evm_compatibility/overview.md:149]: @anton-lisanin So, the error they will see is Bump Allocator out of Memory?

image

File: docs/evm_compatibility/precompiles.md (hunk 11 above)
Comment (m4sterbunny) [reply docs/evm_compatibility/precompiles.md:24]: @YYYurchenko do you have better alt location for the precompiles?

File: docs/evm_compatibility/json_rpc_api_methods.md (hunk 1 above)
Comment (m4sterbunny) [reply docs/evm_compatibility/json_rpc_api_methods.md:12]: @afalaleev pls also confirm you are happy with the "to-do" items -- I think some of these should have a different status.

File: docs/evm_compatibility/overview.md (hunk 9 above)
Comment (m4sterbunny) [reply docs/evm_compatibility/overview.md]: @afalaleev can we have a few lines on why map good list bad (if that is the correct logic?!)?

File: docs/evm_compatibility/overview.md (hunk 10 above)
Comment (m4sterbunny) [reply docs/evm_compatibility/overview.md]: Structure required to support end user:

- Describe Solana accounts structure
- Describe how nEVM uses accounts/ tx counter / data storage etc
{special architecture specific to nEVM -- no hash table as per Eth-EVM}
- How contract execution data is managed and held
- How Solidity contract compiler deals with batches of data in Solana accounts

**Review by Vedad84 — APPROVED** [review]
The existing opcodes are correct to the best of my knowledge.  However, the updated opcodes have an additional following code that is not part of this PR:

GAS(gas left in the transaction) - returns transaction gas limit

**Pull request merged**